RoddyCore/src/de/dkfz/roddy/client/cliclient/CommandLineCall.groovy

```diff
@@ -6,135 +6,248 @@
+        if (mode.isPresent()) {
+            return mode.get()
+        } else {
+            logger.postAlwaysInfo("The startupmode '${putativeModeSpec}' is not known.")
+            return help
         }
+    }
 
-        //Parse options
+    @CompileDynamic
+    private static Optional<Parameter> processParameter(final RoddyStartupOptions option, final ArbitraryParameter parameter, final List<String> errors) {
+        throw new RuntimeException("Oops! Parsed ArbitraryParameter in 'parseOptions()'")
+    }
+
+    @CompileDynamic
+    private static Optional<Parameter> processParameter(final RoddyStartupOptions option, final ParameterWithValue parameter, final List<String> errors) {
+        // If the option is known ensure the option parsed as with value is indeed accepting values.
+        if (option.acceptsParameters) {
+            Optional.of(parameter)
+        } else {
+            errors << "The option " + option + " is malformed! Parameter value required."
+            Optional.empty()
+        }
+    }
+
+    @CompileDynamic
+    private static Optional<Parameter> processParameter(final RoddyStartupOptions option, final ParameterWithoutValue parameter, final List<String> errors) {
+        // If the option is known ensure the option parsed as with value is indeed accepting values.
+        if (!option.acceptsParameters) {
+            Optional.of(parameter)
+        } else {
+            errors << "The option " + option + " is malformed! No parameter value expected."
+            Optional.empty()
+        }
+    }
+
+    @CompileDynamic
+    private static Optional<Parameter> processParameter(final RoddyStartupOptions option, final CValueParameter parameter, final List<String> errors) {
+        Optional.of(parameter)
+    }
+
+    /** Check whether the parsed option is indeed accepted as option by Roddy. Errors are accumulated in the error argument.
+     *  If it is an accepted option, add the option value. Errors are accumulated in the error variable. */
+    @CompileDynamic
+    private static Optional<MapEntry> handle(final Parameter parameter, final List<String> errors) {
+        Optional<RoddyStartupOptions> res = RoddyStartupOptions.fromString(parameter.name)
+        if (!res.isPresent()) {
+            // Add error message, for unknown options
+            errors << "Unknown option '" + parameter.name + "'"
+            Optional.empty()
+        } else {
+            RoddyStartupOptions option = res.get() as RoddyStartupOptions
+            processParameter(option, parameter, errors).
+                    map { new MapEntry(option, it) }
+        }
+    }
+
+
+    /** This method exists only because java-petitparser apparently has no notion of a fatally failing parse expression with or(). Instead it simply
+     *  continues to parse with the next alternative and ignores the fatal error. Maybe there is a workaround using a different grammar.
```

Comment:
Eventually, the parser fails with "end of input expected", because that is expected in the last option. Maybe there is a smarter way with and(), not(), neg(), or() parser combinators in petitparser that I haven't figured out (and postponed, to get this finished).